### PR TITLE
Change NUsense to NUSense, NUSenseCpp to NUSense, some typo fixes, formatting, other stylistic changes

### PR DESCRIPTION
**Pull request by ysims**
Reimport project NUSense and remove the NUSenseCpp from the IDE, then build should be fine.

All remaining warnings are from auto-generated files.

Pretty sure auto-generated are not touched, but not sure if RS485 is auto-generated? It doesn't say. If so can just include it in the formatter exclude list. Auto-generated files don't seem to be easily separable except the whole folders, but it looks like the ones that aren't ours are C, so I've limited it to the `cpp` and `hpp` files. Not sure if this properly separates them.

`format.py` can now be run to format the codebase. Use the `--all` flag to format all files (subject to inclusions/exclusions specified in the script).

Has a lot of files changed but most of them are renaming the root folder.

### CODE REVIEW

**Review by Claegtun — DISMISSED** [review]
Note that we may want to add licences to files that we have made. However, that can be done as a separate PR.

File: NUSense/.cproject [hunk 1]
```diff
@@ -240,10 +240,10 @@
...
 		<configuration configurationName="Debug">
-			<resource resourceType="PROJECT" workspacePath="/NUsenseCpp"/>
+			<resource resourceType="PROJECT" workspacePath="/NUSense"/>
 		</configuration>
 		<configuration configurationName="Release">
-			<resource resourceType="PROJECT" workspacePath="/NUsenseCpp"/>
+			<resource resourceType="PROJECT" workspacePath="/NUSense"/>
 		</configuration>
 	</storageModule>
 </cproject>
```
Comment (JohanneMontano) [inline NUSense/.cproject:249]: Consider adding a newline so git doesnt cry about it

File: NUSense/Core/Inc/imu.h [hunk 2]
```diff
@@ -721,6 +721,6 @@ namespace NUsense {
     extern uint8_t SPI_flag;
     extern uint8_t INT_flag;
 
-}  // namespace NUsense
+}  // namespace NUSense
 
 #endif  //_IMU_H_
```
Comment (JohanneMontano) [inline NUSense/Core/Inc/imu.h:726]: Same as above

File: NUSense/Core/Src/platform/NUSense/NUSenseIO/startup.cpp [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+        /*
+            ~~~ ~~~ ~~~ Basic Set-up ~~~ ~~~ ~~~
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            Here, the servos are set up to have no return-delay, to return always
+            to a write-instruction (unlike the OpenCR set-up), and to have time-
+            based profile-velocity control.
+        */
+
+        // For each port, write for all servos the return-delay-time to be 0 μs.
+        for (int i = 0; i < NUM_PORTS; i++) {
```
Comment (JohanneMontano) [inline NUSense/Core/Src/platform/NUSense/NUSenseIO/startup.cpp:22]: Future note,it might be ideal to do these in one loop if we can

Comment (JohanneMontano) [inline NUSense/Core/Src/platform/NUSense/NUSenseIO/startup.cpp:22]: Might wanna do these in one for loop in the future if possible

**Comment by JohanneMontano** [comment]
Does this mean we can now put branch protections and formatting rules like we do in NUbots?!?!?! :rocket:  :partying_face:

File: NUSense/Core/Src/platform/NUSense/NUSenseIO/startup.cpp (hunk 3 above)
Comment (ysims) [reply NUSense/Core/Src/platform/NUSense/NUSenseIO/startup.cpp:22]: depends if order matters

**Review by Claegtun — APPROVED** [review]
Seems good.

**Pull request merged**